inc/az_ulib_descriptor_api.h

```diff
@@ -64,37 +72,72 @@ typedef struct az_ulib_capability_descriptor_tag
 } az_ulib_interface_descriptor;
 
+/**
+ * @brief   Create a new Interface descriptor.
+ *
+ * This API creates a new description for a interface, it will create the
+ * #az_ulib_interface_descriptor with the provided interface name and version, adding all the
+ * capabilities.
+ *
+ * The descriptor will be stored in the Text area (Flash) that will be pointed by the provided
+ * interface_var. In this way, no other components on the system needs to copy any of the data on
+ * the descriptor.
+ *
+ * @param[in]   interface_name    The `/0` terminated `const char* const` with the interface name.
+ *                                It cannot be `NULL` and shall be allocated in a way that it stays
+ *                                valid until the interface is unpublished at some (potentially)
+ *                                unknown time in the future.
+ * @param[in]   interface_version The #az_ulib_version with the interface version.
+ * @param[in]   capability_size   The number of #az_ulib_capability_descriptor in the interface.
+ * @param[in]   capabilities      The list of #az_ulib_capability_descriptor in the interface.
+ * @return The #az_ulib_interface_descriptor with the description of the interface.
+ */
+#define AZ_ULIB_DESCRIPTOR_CREATE(                                                  \
+    interface_name, interface_version, capability_size, capabilities)               \
+  {                                                                                 \
+    .name = AZ_SPAN_LITERAL_FROM_STR(interface_name), .version = interface_version, \
+    .size = capability_size, .capability_list = capabilities                        \
+  }
+
 /**
  * @brief   Add property to the interface descriptor.
  *
  * Populate a new [*property* capability](#AZ_ULIB_CAPABILITY_TYPE_PROPERTY) to add to the
  * interface.
  *
- * @param[in]   name        The `/0` terminated `const char* const` with the property name. It
- *                          cannot be `NULL` and shall be allocated in a way that it stays valid
- *                          until the interface is unpublished at some (potentially) unknown time
- *                          in the future.
- * @param[in]   get         The function pointer to #az_ulib_capability_get with the implementation
- *                          of the get command for the property. The get command shall be valid
- *                          until the interface is unpublished at some (potentially) unknown time
- *                          in the future.
- * @param[in]   set         The function pointer to #az_ulib_capability_set with the implementation
- *                          of the set command for the property. The set command shall be valid
- *                          until the interface is unpublished at some (potentially) unknown time
- *                          in the future.
+ * @param[in]   property_name The `/0` terminated `az_span` with the property name. It cannot be
```

Comment:
I believe this is still technically a `const char* const` when it is passed to this macro. It is stored as a span but from user's perspective it's a `char*`

Reply:
Is it standard to dive into the types wrapped by typedefs like this in documentation? I feel like it would be appropriate to leave it to the user to see what the typedef expands to.

Reply:
I agree with Dane, the argument shall be `const char* const` once the conversion is internal of the macro. 

---
In reply to: [600710541](https://github.com/Azure/azure-ulib-c/pull/63#discussion_r600710541) [](ancestors = 600710541)